File: level76/classwork/codwars3.py
#https://www.codewars.com/kata/526a569ca578d7e6e300034e/train/python
def to_decimal(input, alphabet):
    result = 0
    for p,d in enumerate(reversed(input)):
        result += alphabet.index(d)*len(alphabet)**p
    
    return result
    
def to_alphabet(input, alphabet):
    result = ""
    while input >= len(alphabet):
        result += alphabet[input % len(alphabet)]
        input = int(input / len(alphabet))
    
    result += alphabet[input]
    return "".join(reversed(result))

def convert(input, source, target):
    dec = to_decimal(input, source)
    final = to_alphabet(dec, target)
    return final

File: level 076/classwork/codwars3.js
//https://www.codewars.com/kata/526a569ca578d7e6e300034e/train/javascript
function convert(input, source, target) {
  let s=0;  let str='';
  for (let i=0; i<input.length; i++) {
    s=s*source.length+source.indexOf(input[i]);
  }
  while (s>0) {
    str=target[s%target.length]+str;
    s=Math.floor(s/target.length);
  }  
  return str ? str : target[0];
}
